### PR TITLE
Add Jboss eap clustered multivm rhel

### PR DESCRIPTION
**Pull request by Suraj2093**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add template and all files required for JBoss EAP clustered MultiVM RHEL

### CODE REVIEW

File: jboss-eap-clustered-multivm-rhel/scripts/jbosseap-setup-redhat.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+echo "Red Hat JBoss EAP Cluster Intallation Start " | adddate >> jbosseap.install.log
+/bin/date +%H:%M:%S  >> jbosseap.install.log
+
+echo 'export EAP_HOME="/opt/rh/eap7/root/usr/share"' >> ~/.bash_profile
+source ~/.bash_profile
+touch /etc/profile.d/eap_env.sh
+echo 'export EAP_HOME="/opt/rh/eap7/root/usr/share"' >> /etc/profile.d/eap_env.sh
+
+while getopts "a:t:p:f:" opt; do
```
Comment (bmoore-msft) [inline jboss-eap-clustered-multivm-rhel/scripts/jbosseap-setup-redhat.sh:17]: why do we have this for only some of the parameters?  some are positional, some not...

File: jboss-eap-clustered-multivm-rhel/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+      "defaultValue": "password",
+      "allowedValues": [
+        "password",
+        "sshPublicKey"
+      ],
+      "metadata": {
+        "description": "Type of authentication to use on the Virtual Machine"
+      }
+    },
+    "adminPasswordOrSSHKey": {
```
Comment (bmoore-msft) [inline jboss-eap-clustered-multivm-rhel/azuredeploy.json:22]: format the json, e.g. shift+alt+f in VS code

File: jboss-eap-clustered-multivm-rhel/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+            "type": "CustomScript",
+            "typeHandlerVersion": "2.0",
+            "autoUpgradeMinorVersion": true,
+            "settings": {
+              "fileUris": [
+                "[uri(parameters('_artifactsLocation'), concat('scripts/jbosseap-setup-redhat.sh', parameters('_artifactsLocationSasToken')))]"
+              ]
+            },
+            "protectedSettings": {
+              "commandToExecute": "[concat('sh jbosseap-setup-redhat.sh',' ',variables('scriptArgs'),' ',parameters('jbossEAPUserName'),' ',variables('singlequote'),parameters('jbossEAPPassword'),variables('singlequote'),' ',parameters('rhelOSLicenseType'),' ',parameters('rhsmUserName'),' ',variables('singlequote'),parameters('rhsmPassword'),variables('singlequote'),' ', parameters('rhsmPoolEAP'),' ',variables('storageAccountName'),' ',variables('containerName'),' ',base64(listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2019-04-01').keys[0].value), ' ', parameters('eapOnRHELVersion'),' ', parameters('rhsmPoolRHEL'))]"
```
Comment (bmoore-msft) [inline jboss-eap-clustered-multivm-rhel/azuredeploy.json:384]: see the changes made to the VMSS template and make the same changes here

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by bmoore-msft** [comment]
Be sure to make appropriate changes in #8200 as well

File: jboss-eap-clustered-multivm-rhel/scripts/jbosseap-setup-redhat.sh (hunk 1 above)
Comment (Suraj2093) [reply jboss-eap-clustered-multivm-rhel/scripts/jbosseap-setup-redhat.sh:17]: This is to download the WAR file the same way we handle the script itself - by downloading from the staged location. I have followed the same method from https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/scripts/copyfilefromazure.sh as you mentioned earlier for the WildFly on CentOS template https://github.com/Azure/azure-quickstart-templates/pull/7961#discussion_r472398489

File: jboss-eap-clustered-multivm-rhel/azuredeploy.json (hunk 2 above)
Comment (Suraj2093) [reply jboss-eap-clustered-multivm-rhel/azuredeploy.json:22]: I have formatted the whole template now.

File: jboss-eap-clustered-multivm-rhel/azuredeploy.json (hunk 3 above)
Comment (Suraj2093) [reply jboss-eap-clustered-multivm-rhel/azuredeploy.json]: I have replicated the changes from VMSS template here and the standalone template as well. Tested them both successfully.

**Pull request merged**